aztec_backend_wasm/src/lib.rs

```diff
@@ -42,32 +44,46 @@ pub fn compute_witnesses(
     // Now use the partial witness generator to fill in the rest of the witnesses
     // which are possible
 
+    use barretenberg_wasm::Plonk;
     let plonk = Plonk;
+    let num_witnesses = circuit.num_vars();
     match plonk.solve(&mut witness_map, circuit.opcodes) {
         Ok(_) => {}
         Err(opcode) => panic!("solver came across an error with opcode {}", opcode),
     };
 
-    // Serialize the witness in a way that the C++ codebase can deserialize
-    let assignments = proof::flatten_witness_map(&circuit, witness_values);
+    // Add witnesses in the correct order
+    // Note: The witnesses are sorted via their witness index
+    // witness_values may not have all the witness indexes, e.g for unused witness which are not solved by the solver
+    let mut sorted_witness = common::barretenberg_structures::Assignments::new();
+    for i in 1..num_witnesses {
+        // Get the value if it exists. If i does not, then we fill it with the zero value
+        let value = match witness_map.get(&Witness(i)) {
+            Some(value) => *value,
+            None => FieldElement::zero(),
+        };
+
+        sorted_witness.push(value);
+    }
 
-    assignments.to_bytes()
+    sorted_witness.to_bytes()
```

Comment:
This change needs to be reverted. This logic has been factored out into `proof::flatten_witness_map`